leetcode/1846/solution.go

```diff
@@ -0,0 +1,37 @@
+// https://leetcode.com/problems/maximum-element-after-decreasing-and-rearranging/
+//
+// The problem can be easily solved by O(nlogn) time complexity.
+// Below is an overcomplicated solution with O(n) + O(mlogm) time complexity where m is the unique number of the array.
```

Comment:
Does O(n) + O(mlogm) = O(nlogn)?  
Because m < n, I am not sure I am right, just a guess.

Reply:
You are very right. In the worst case where each num is unique, it's just O(nlogn). That's why I put it overcomplicated.

Reply:
You are very right. In the worst case where each num is unique, it's just O(nlogn). That's why I put it overcomplicated.

Reply:
Thanks for reviewing BTW